src/rrd_xport.c

```diff
@@ -444,15 +444,17 @@ int rrd_graph_xport(image_desc_t *im) {
   default:
     break;
   }
+
+  /* free legend */
+  for (unsigned long j = 0; j < col_cnt; j++) {
+    free(legend_v[j]);
+  }
+  free(legend_v);
+  /* free data */
+  free(data);
+
```

Comment:
Actually if (r != 0), free(data) is done and then return; however "data" is allocated by rrd_xport_fn() (line 407), (r==0) (switch(im->imgformat) case default)  but not free'd without this patch.